# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/gnocl/rebuild_safe.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- gnocl-0.9.96/src/install.tcl.vanilla	2025-08-26 17:46:06.603887896 +0200
+++ gnocl-0.9.96/src/install.tcl	2025-08-26 17:47:08.268777192 +0200
@@ -51,10 +51,6 @@
 set destDir [file join $dir $name$version]
 switch -- $what {
    "install"   {
-       if { [file exists $destDir] } {
-           puts "$destDir exists already. Aborting installation."
-           exit -1
-           }
            puts "Create updated package file."
            set fp [open pkgIndex.tcl "w"]
            puts $fp "# Created: [date] [date time]"
@@ -70,12 +66,12 @@
            puts $fp "package ifneeded $package $version \[list load \[file join \$dir $name.so\]\]"
            close $fp
            puts "Creating $destDir"
-           file mkdir $destDir
+           file mkdir -p $destDir
            set files [glob *.so]
            lappend files pkgIndex.tcl
            foreach file $files {
                puts "Copying $file"
-               file copy $file $destDir
+               file copy -force $file $destDir
                exec chmod 444 $file
                }
         }
